{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#FASTQC and trimming info here: http://onsnetwork.org/kubu4/2018/05/16/trimgalorefastqcmultiqc-trimgalore-rrbs-geoduck-bs-seq-fastq-data-directional/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir /Volumes/web/metacarcinus/Pgenerosa/GENOMES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070/Bismark_genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#copy Pgenerosa genome v070 to metacarcinus folder to build genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cp /Volumes/web-1/halfshell/genomic-databank/Pgenerosa_v070.fa /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#move to the directory where we want to save the genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cd /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#build Bismark genome using ostrich"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!/Users/Shared/bioinformatics/bismark_v0.15.0/bismark_genome_preparation \\\n",
    "--path_to_bowtie /Users/Shared/bioinformatics/bowtie2-2.2.9//bowtie2 \\\n",
    "--verbose /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070 \\\n",
    "2> /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070/genomeprep.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#genomeprep.err says \"Warning: Encountered reference sequence with only gaps\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Checking .fa file for gaps and missing sequence (http://seqanswers.com/forums/showthread.php?t=6301)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#first make short file with only the chromosome and the first line of sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!grep '>'  -A 1 /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070/Pgenerosa_v070.fa \\\n",
    "> /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070/missing_seq_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#then check for chromosomes where there is no sequence (based on this code:  https://www.biostars.org/p/78786/)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">"
     ]
    }
   ],
   "source": [
    "!awk 'BEGIN {RS = \">\" ; FS = \"\\n\" ; ORS = \"\"} {if (!$2) print \">\"$0}' \\\n",
    "/Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070/missing_seq_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#doesn't seem to be any chromosomes where sequence is missing \n",
    "#not sure why the carrot printed out\n",
    "#running another check in case there is a line with no chromosome name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!grep '>$' /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070/missing_seq_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#confirmed there are no chromosomes without any sequence or random carrots\n",
    "#now testing for NNNNs in seqeuence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GNNNNNNNNNNAACTGTTACACAATGCATCATCCGAGGATCTTTTGTAATAATAAGTATCACCCTCCTGACTCAGACCAA\n",
      "TTTTTTCTCCCACCCGGAATGTTTTCTATCCCCTTCCCCCCCNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCAAACAGCATGGGAACAA\n",
      "TACATTAGGAACTTTGCTATAAAGGACAGTAAGATAATTGACAAAAATGTACAGATGTATACTTAAAAGTNNNNNNNNNN\n",
      "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "NNNNNNNNNNGACATGTTCCCTACTGTCATTTCCAGATCTGTGAGACATGTTCCCTAATGTCATTTCCATCATTGTGAGA\n",
      "ACCGGTAGAGGGTACTGCATAGAATAATGTTCAGAATATATATANNNNNNNNNNGATACCGGTAGAGGGTACTGCATAGA\n",
      "TTGCATAGGAGACTAGACAGAAGATTTTTTTTTTCGATCTGAAATTCAAGATGGCCGCCATTACTAAAAATNNNNNNNNN\n",
      "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "NNNNNNNNNNNATGTAAATGATTTTGCCTATATCTTCACAGAAGCATTGTTTGGTGAAGGAGGTTCAGATTTGTAAAAAT\n",
      "^C\n"
     ]
    }
   ],
   "source": [
    "!grep NNNN /Volumes/web/metacarcinus/Pgenerosa/GENOMES/v070/Pgenerosa_v070.fa |head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#lots of lines contain many Ns. Is this problematic? No, see issue on GitHub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#test alignments using subset of data. \n",
    "#Ran 20181011_bmrkAS1.2I60test_Pgenr.sh on mox which aligned 10K reads per sample\n",
    "#transferred everything from gscratch on mox to Gannet\n",
    "#rsync --archive --progress --verbose strigg@mox.hyak.uw.edu:/gscratch/srlab/strigg/analyses/20181011 ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#report mapping efficiency and genome coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create summary report of mapping efficiency and dedup. \n",
    "!cat /Volumes/web/metacarcinus/Pgenerosa/20181011/*PE_report.txt | \\\n",
    "grep 'Mapping\\ efficiency\\:' | \\\n",
    "cat - /Volumes/web/metacarcinus/Pgenerosa/20181011/*.deduplication_report.txt | \\\n",
    "grep 'Mapping\\ efficiency\\:\\|removed' \\\n",
    "> /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI_mapping_dedup_summary.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      " *** Bismark methylation extractor version v0.20.0 ***\n",
      "\n",
      "Output will be written into the directory: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "Trying to determine the type of mapping from the SAM header line of file /Volumes/web/metacarcinus/Pgenerosa/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "samtools view: failed to open \"/Volumes/web/metacarcinus/Pgenerosa/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\" for reading: No such file or directory\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/web/metacarcinus/Pgenerosa/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
      "Captured error message: '[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "\n",
      "[ERROR] The file appears to be truncated, please ensure that there were no errors while copying the file!!! Exiting...\n",
      "\n",
      "\n",
      " *** Bismark methylation extractor version v0.20.0 ***\n",
      "\n",
      "Output will be written into the directory: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "Trying to determine the type of mapping from the SAM header line of file /Volumes/web/metacarcinus/Pgenerosa/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "samtools view: failed to open \"/Volumes/web/metacarcinus/Pgenerosa/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\" for reading: No such file or directory\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/web/metacarcinus/Pgenerosa/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
      "Captured error message: '[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "\n",
      "[ERROR] The file appears to be truncated, please ensure that there were no errors while copying the file!!! Exiting...\n",
      "\n",
      "\n",
      " *** Bismark methylation extractor version v0.20.0 ***\n",
      "\n",
      "Output will be written into the directory: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "Trying to determine the type of mapping from the SAM header line of file /Volumes/web/metacarcinus/Pgenerosa/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "samtools view: failed to open \"/Volumes/web/metacarcinus/Pgenerosa/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\" for reading: No such file or directory\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/web/metacarcinus/Pgenerosa/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
      "Captured error message: '[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "\n",
      "[ERROR] The file appears to be truncated, please ensure that there were no errors while copying the file!!! Exiting...\n",
      "\n",
      "\n",
      " *** Bismark methylation extractor version v0.20.0 ***\n",
      "\n",
      "Output will be written into the directory: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "Trying to determine the type of mapping from the SAM header line of file /Volumes/web/metacarcinus/Pgenerosa/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "samtools view: failed to open \"/Volumes/web/metacarcinus/Pgenerosa/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\" for reading: No such file or directory\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/web/metacarcinus/Pgenerosa/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
      "Captured error message: '[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "\n",
      "[ERROR] The file appears to be truncated, please ensure that there were no errors while copying the file!!! Exiting...\n",
      "\n",
      "\n",
      " *** Bismark methylation extractor version v0.20.0 ***\n",
      "\n",
      "Output will be written into the directory: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "Trying to determine the type of mapping from the SAM header line of file /Volumes/web/metacarcinus/Pgenerosa/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "samtools view: failed to open \"/Volumes/web/metacarcinus/Pgenerosa/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\" for reading: No such file or directory\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/web/metacarcinus/Pgenerosa/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
      "Captured error message: '[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "\n",
      "[ERROR] The file appears to be truncated, please ensure that there were no errors while copying the file!!! Exiting...\n",
      "\n",
      "\n",
      " *** Bismark methylation extractor version v0.20.0 ***\n",
      "\n",
      "Output will be written into the directory: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "Trying to determine the type of mapping from the SAM header line of file /Volumes/web/metacarcinus/Pgenerosa/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "samtools view: failed to open \"/Volumes/web/metacarcinus/Pgenerosa/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\" for reading: No such file or directory\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/web/metacarcinus/Pgenerosa/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
      "Captured error message: '[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "\n",
      "[ERROR] The file appears to be truncated, please ensure that there were no errors while copying the file!!! Exiting...\n",
      "\n",
      "\n",
      " *** Bismark methylation extractor version v0.20.0 ***\n",
      "\n",
      "Output will be written into the directory: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "Trying to determine the type of mapping from the SAM header line of file /Volumes/web/metacarcinus/Pgenerosa/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "samtools view: failed to open \"/Volumes/web/metacarcinus/Pgenerosa/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\" for reading: No such file or directory\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/web/metacarcinus/Pgenerosa/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
      "Captured error message: '[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "\n",
      "[ERROR] The file appears to be truncated, please ensure that there were no errors while copying the file!!! Exiting...\n",
      "\n",
      "\n",
      " *** Bismark methylation extractor version v0.20.0 ***\n",
      "\n",
      "Output will be written into the directory: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "Trying to determine the type of mapping from the SAM header line of file /Volumes/web/metacarcinus/Pgenerosa/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "samtools view: failed to open \"/Volumes/web/metacarcinus/Pgenerosa/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\" for reading: No such file or directory\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/web/metacarcinus/Pgenerosa/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
      "Captured error message: '[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "\n",
      "[ERROR] The file appears to be truncated, please ensure that there were no errors while copying the file!!! Exiting...\n",
      "\n",
      "\n",
      " *** Bismark methylation extractor version v0.20.0 ***\n",
      "\n",
      "Output will be written into the directory: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "Trying to determine the type of mapping from the SAM header line of file /Volumes/web/metacarcinus/Pgenerosa/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "samtools view: failed to open \"/Volumes/web/metacarcinus/Pgenerosa/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\" for reading: No such file or directory\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: /Volumes/web/metacarcinus/Pgenerosa/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/web/metacarcinus/Pgenerosa/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
      "Captured error message: '[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "\n",
      "[ERROR] The file appears to be truncated, please ensure that there were no errors while copying the file!!! Exiting...\n",
      "\n",
      "\n",
      " *** Bismark methylation extractor version v0.20.0 ***\n",
      "\n",
      "Output will be written into the directory: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "Trying to determine the type of mapping from the SAM header line of file /Volumes/web/metacarcinus/Pgenerosa/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "samtools view: failed to open \"/Volumes/web/metacarcinus/Pgenerosa/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\" for reading: No such file or directory\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/web/metacarcinus/Pgenerosa/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
      "Captured error message: '[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "\n",
      "[ERROR] The file appears to be truncated, please ensure that there were no errors while copying the file!!! Exiting...\n",
      "\n",
      "\n",
      " *** Bismark methylation extractor version v0.20.0 ***\n",
      "\n",
      "Output will be written into the directory: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "Trying to determine the type of mapping from the SAM header line of file /Volumes/web/metacarcinus/Pgenerosa/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "samtools view: failed to open \"/Volumes/web/metacarcinus/Pgenerosa/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\" for reading: No such file or directory\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/web/metacarcinus/Pgenerosa/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
      "Captured error message: '[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "\n",
      "[ERROR] The file appears to be truncated, please ensure that there were no errors while copying the file!!! Exiting...\n",
      "\n",
      "\n",
      " *** Bismark methylation extractor version v0.20.0 ***\n",
      "\n",
      "Output will be written into the directory: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "Trying to determine the type of mapping from the SAM header line of file /Volumes/web/metacarcinus/Pgenerosa/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "samtools view: failed to open \"/Volumes/web/metacarcinus/Pgenerosa/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\" for reading: No such file or directory\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/web/metacarcinus/Pgenerosa/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
      "Captured error message: '[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "\n",
      "[ERROR] The file appears to be truncated, please ensure that there were no errors while copying the file!!! Exiting...\n",
      "\n",
      "\n",
      " *** Bismark methylation extractor version v0.20.0 ***\n",
      "\n",
      "Output will be written into the directory: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "Trying to determine the type of mapping from the SAM header line of file /Volumes/web/metacarcinus/Pgenerosa/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "[E::hts_open_format] Failed to open file /Volumes/web/metacarcinus/Pgenerosa/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "samtools view: failed to open \"/Volumes/web/metacarcinus/Pgenerosa/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam\" for reading: No such file or directory\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Number of cores to be used: 1\n",
      "Output path specified as: /Volumes/web/metacarcinus/Pgenerosa/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "^C\r\n"
     ]
    }
   ],
   "source": [
    "#run methylation extractor\n",
    "#This command isn't working yet...do I even need to run this?\n",
    "!find /Volumes/web/metacarcinus/Pgenerosa/20181011/*.deduplicated.bam \\\n",
    "| xargs basename -s _R1_001_val_1_bismark_bt2_pe.deduplicated.bam | \\\n",
    "xargs -I{} bismark_methylation_extractor --gzip -bedGraph \\\n",
    "/Volumes/web/metacarcinus/Pgenerosa/{}_R1_001_val_1_bismark_bt2_pe.deduplicated.bam \\\n",
    "-o /Volumes/web/metacarcinus/Pgenerosa/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "##compiling bam files and sorting"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "find /Volumes/web/metacarcinus/Pgenerosa/20181011/*deduplicated.bam| \\\n",
    "xargs basename -s _R1_001_val_1_bismark_bt2_pe.deduplicated.bam | xargs -I{} /Users/Shelly/anaconda3/bin/samtools \\\n",
    "sort /Volumes/web/metacarcinus/Pgenerosa/20181011/{}_R1_001_val_1_bismark_bt2_pe.deduplicated.bam \\\n",
    "-o /Volumes/web/metacarcinus/Pgenerosa/20181011/{}_dedup.sorted.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## create methylation extractor files for estimating coverage. \n",
    "## got command from Steven's notebook (https://github.com/sr320/nb-2018/blob/master/O_lurida/07-Bismark-wgbs.ipynb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file /Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OT_EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOT_EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_CTOB_EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/web/metacarcinus/Pgenerosa/20181011/CpG_OB_EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Users/Shared/bioinformatics/bismark_v0.15.0/bismark_methylation_extractor \\\n",
    "--paired-end --bedGraph --counts --scaffolds \\\n",
    "--multicore 2 \\\n",
    "/Volumes/web/metacarcinus/Pgenerosa/20181011/*deduplicated.bam \\\n",
    "-o /Volumes/web/metacarcinus/Pgenerosa/20181011/ \\\n",
    "--samtools /usr/local/bin/samtools \\\n",
    "2> /Volumes/web/metacarcinus/Pgenerosa/20181011/bme.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#make summary report \n",
    "#need to install latestversion of bismark because 0.15.0 doesn't ahve summary option\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /Users/strigg/anaconda3\n",
      "\n",
      "  added / updated specs: \n",
      "    - bismark\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    psutil-5.4.7               |   py36h1de35cc_0         314 KB\n",
      "    decorator-4.3.0            |           py36_0          15 KB\n",
      "    parso-0.3.1                |           py36_0         114 KB\n",
      "    partd-0.3.9                |           py36_0          32 KB\n",
      "    pylint-2.1.1               |           py36_0         795 KB\n",
      "    datashape-0.5.4            |           py36_1         100 KB\n",
      "    pep8-1.7.1                 |           py36_0          51 KB\n",
      "    qtawesome-0.5.1            |           py36_1         569 KB\n",
      "    unicodecsv-0.14.1          |   py36he531d66_0          25 KB\n",
      "    conda-build-3.10.9         |           py36_0         433 KB\n",
      "    pycparser-2.19             |           py36_0         173 KB\n",
      "    pytables-3.4.4             |   py36h13cba08_0         1.4 MB\n",
      "    kiwisolver-1.0.1           |   py36h0a44026_0          56 KB\n",
      "    backports.os-0.1.1         |           py36_0          15 KB\n",
      "    isort-4.3.4                |           py36_0          58 KB\n",
      "    h5py-2.8.0                 |   py36h878fce3_3         913 KB\n",
      "    jupyter_client-5.2.3       |           py36_0         124 KB\n",
      "    pyyaml-3.13                |   py36h1de35cc_0         160 KB\n",
      "    xlwt-1.2.0                 |   py36h5ad1178_0         163 KB\n",
      "    pkginfo-1.4.2              |           py36_1          39 KB\n",
      "    qtpy-1.5.2                 |           py36_0          53 KB\n",
      "    python.app-2               |           py36_9         1.3 MB\n",
      "    defusedxml-0.5.0           |           py36_1          29 KB\n",
      "    bowtie2-2.3.4.3            |   py36h2d50403_0         2.0 MB  bioconda\n",
      "    itsdangerous-1.0.0         |           py36_0          24 KB\n",
      "    pyzmq-17.1.2               |   py36h1de35cc_0         407 KB\n",
      "    zict-0.1.3                 |           py36_0          18 KB\n",
      "    flask-1.0.2                |           py36_1         119 KB\n",
      "    tornado-5.1.1              |   py36h1de35cc_0         662 KB\n",
      "    astroid-2.0.4              |           py36_0         247 KB\n",
      "    ipython-7.0.1              |   py36h39e3cac_0         1.0 MB\n",
      "    pytest-arraydiff-0.2       |   py36h39e3cac_0          14 KB\n",
      "    backports-1.0              |           py36_1           4 KB\n",
      "    clyent-1.2.2               |           py36_1          19 KB\n",
      "    chardet-3.0.4              |           py36_1         189 KB\n",
      "    numpy-1.15.3               |   py36h6a91979_0          35 KB\n",
      "    numpy-base-1.15.3          |   py36h8a80b8c_0         4.0 MB\n",
      "    llvmlite-0.25.0            |   py36h8c7ce04_0        10.7 MB\n",
      "    sip-4.19.8                 |   py36h0a44026_0         252 KB\n",
      "    imagesize-1.1.0            |           py36_0           9 KB\n",
      "    seaborn-0.9.0              |           py36_0         379 KB\n",
      "    pyqt-5.9.2                 |   py36h655552a_2         4.4 MB\n",
      "    conda-verify-3.1.1         |           py36_0          33 KB\n",
      "    jedi-0.13.1                |           py36_0         229 KB\n",
      "    distributed-1.23.3         |           py36_0         819 KB\n",
      "    openpyxl-2.4.0             |           py36_0         268 KB  bioconda\n",
      "    mkl-service-1.1.2          |   py36h6b9c3cc_5           9 KB\n",
      "    testpath-0.4.2             |           py36_0          91 KB\n",
      "    matplotlib-3.0.0           |   py36h54f8f79_0         6.8 MB\n",
      "    atomicwrites-1.2.1         |           py36_0          11 KB\n",
      "    pygments-2.2.0             |   py36h240cd3f_0         1.3 MB\n",
      "    scipy-1.1.0                |   py36h28f7352_1        15.4 MB\n",
      "    statsmodels-0.9.0          |   py36h1d22016_0         8.5 MB\n",
      "    appdirs-1.4.3              |   py36h28b3542_0          16 KB\n",
      "    appnope-0.1.0              |   py36hf537a9a_0           8 KB\n",
      "    jdcal-1.4                  |           py36_0          11 KB\n",
      "    cython-0.29                |   py36h0a44026_0         2.1 MB\n",
      "    jupyterlab_server-0.2.0    |           py36_0          40 KB\n",
      "    lxml-4.2.5                 |   py36hef8c89e_0         1.5 MB\n",
      "    conda-4.5.11               |           py36_0         1.0 MB\n",
      "    sortedcontainers-2.0.5     |           py36_0          42 KB\n",
      "    ipywidgets-7.4.2           |           py36_0         151 KB\n",
      "    jupyterlab-0.35.2          |           py36_0        10.5 MB\n",
      "    anaconda-client-1.7.2      |           py36_0         141 KB\n",
      "    requests-2.14.2            |           py36_0         720 KB\n",
      "    colorama-0.4.0             |           py36_0          23 KB\n",
      "    webencodings-0.5.1         |           py36_1          19 KB\n",
      "    pycrypto-2.6.1             |   py36h1de35cc_9         452 KB\n",
      "    python-dateutil-2.7.3      |           py36_0         260 KB\n",
      "    nose-1.3.7                 |           py36_2         214 KB\n",
      "    bokeh-1.0.0                |           py36_0         5.3 MB\n",
      "    blaze-0.11.3               |           py36_0         603 KB\n",
      "    jupyter_core-4.4.0         |           py36_0          63 KB\n",
      "    sortedcollections-1.0.1    |           py36_0          15 KB\n",
      "    lazy-object-proxy-1.3.1    |   py36h1de35cc_2          29 KB\n",
      "    msgpack-python-0.5.6       |   py36h04f5b5a_1          89 KB\n",
      "    xlrd-1.1.0                 |           py36_1         194 KB\n",
      "    pyparsing-2.2.2            |           py36_0          97 KB\n",
      "    appscript-1.0.1            |   py36h1de35cc_1         129 KB\n",
      "    alabaster-0.7.12           |           py36_0          17 KB\n",
      "    wcwidth-0.1.7              |   py36h8c6ec74_0          25 KB\n",
      "    mpmath-1.0.0               |           py36_2         893 KB\n",
      "    nltk-3.3.0                 |           py36_0         2.0 MB\n",
      "    patsy-0.5.0                |           py36_0         322 KB\n",
      "    greenlet-0.4.15            |   py36h1de35cc_0          18 KB\n",
      "    constantly-15.1.0          |   py36h28b3542_0          13 KB\n",
      "    nbconvert-5.4.0            |           py36_1         416 KB\n",
      "    zope.interface-4.5.0       |   py36h1de35cc_0         198 KB\n",
      "    ptyprocess-0.6.0           |           py36_0          23 KB\n",
      "    pyopenssl-18.0.0           |           py36_0          82 KB\n",
      "    importlib_metadata-0.6     |           py36_0          21 KB\n",
      "    ply-3.11                   |           py36_0          79 KB\n",
      "    xlwings-0.12.1             |           py36_0         433 KB\n",
      "    pysocks-1.6.8              |           py36_0          22 KB\n",
      "    urllib3-1.12               |           py36_0         151 KB  bioconda\n",
      "    automat-0.7.0              |           py36_0          52 KB\n",
      "    sphinxcontrib-websupport-1.1.0|           py36_1          36 KB\n",
      "    mistune-0.8.4              |   py36h1de35cc_0          54 KB\n",
      "    astropy-3.0.5              |   py36h1de35cc_0         6.7 MB\n",
      "    more-itertools-4.3.0       |           py36_0          83 KB\n",
      "    cloudpickle-0.6.1          |           py36_0          27 KB\n",
      "    keyring-15.1.0             |           py36_0          46 KB\n",
      "    anaconda-navigator-1.9.2   |           py36_0         5.0 MB\n",
      "    rope-0.11.0                |           py36_0         282 KB\n",
      "    pytz-2018.5                |           py36_0         231 KB\n",
      "    ipykernel-5.1.0            |   py36h39e3cac_0         156 KB\n",
      "    qtconsole-4.4.2            |           py36_0         157 KB\n",
      "    pywavelets-1.0.1           |   py36h1d22016_0         4.3 MB\n",
      "    odo-0.5.1                  |   py36hc1af34a_0         193 KB\n",
      "    libdeflate-1.0             |       h470a237_0          44 KB  bioconda\n",
      "    pluggy-0.8.0               |           py36_0          28 KB\n",
      "    babel-2.6.0                |           py36_0         5.7 MB\n",
      "    terminado-0.8.1            |           py36_1          21 KB\n",
      "    cryptography-2.3.1         |   py36hdbc3d79_0         540 KB\n",
      "    asn1crypto-0.24.0          |           py36_0         154 KB\n",
      "    widgetsnbextension-3.4.2   |           py36_0         1.7 MB\n",
      "    flask-cors-3.0.6           |           py36_0          20 KB\n",
      "    cycler-0.10.0              |   py36hfc81398_0          13 KB\n",
      "    scikit-learn-0.20.0        |   py36h4f467ca_1         5.4 MB\n",
      "    certifi-2018.10.15         |           py36_0         138 KB\n",
      "    bkcharts-0.2               |   py36h073222e_0         127 KB\n",
      "    et_xmlfile-1.0.1           |   py36h1315bdc_0          20 KB\n",
      "    twisted-17.5.0             |           py36_0         4.3 MB\n",
      "    nbformat-4.4.0             |   py36h827af21_0         138 KB\n",
      "    mkl_random-1.0.1           |   py36h5d10147_1         349 KB\n",
      "    packaging-18.0             |           py36_0          33 KB\n",
      "    numpydoc-0.8.0             |           py36_0          42 KB\n",
      "    pickleshare-0.7.5          |           py36_0          12 KB\n",
      "    jsonschema-2.6.0           |   py36hb385e00_0          62 KB\n",
      "    multipledispatch-0.6.0     |           py36_0          21 KB\n",
      "    py-1.7.0                   |           py36_0         138 KB\n",
      "    notebook-5.7.0             |           py36_0         7.4 MB\n",
      "    prompt_toolkit-2.0.6       |           py36_0         466 KB\n",
      "    path.py-11.5.0             |           py36_0          55 KB\n",
      "    pyasn1-0.4.4               |   py36h28b3542_0         101 KB\n",
      "    pyodbc-4.0.24              |   py36h0a44026_0          60 KB\n",
      "    sympy-1.3                  |           py36_0         9.5 MB\n",
      "    traitlets-4.3.2            |   py36h65bd3ce_0         131 KB\n",
      "    click-7.0                  |           py36_0         118 KB\n",
      "    toolz-0.9.0                |           py36_0          91 KB\n",
      "    olefile-0.46               |           py36_0          48 KB\n",
      "    pyflakes-2.0.0             |           py36_0          87 KB\n",
      "    wrapt-1.10.11              |   py36h1de35cc_2          41 KB\n",
      "    docutils-0.14              |   py36hbfde631_0         689 KB\n",
      "    entrypoints-0.2.3          |           py36_2           9 KB\n",
      "    bottleneck-1.2.1           |   py36h1d22016_1         113 KB\n",
      "    mccabe-0.6.1               |           py36_1          14 KB\n",
      "    zope-1.0                   |           py36_1           3 KB\n",
      "    filelock-3.0.9             |           py36_0          13 KB\n",
      "    python-3.6.6               |       hc167b69_0        15.4 MB\n",
      "    prometheus_client-0.4.2    |           py36_0          63 KB\n",
      "    bitarray-0.8.3             |   py36h1de35cc_0          54 KB\n",
      "    contextlib2-0.5.5          |   py36hd66e5e7_0          15 KB\n",
      "    locket-0.2.0               |   py36hca03003_1           8 KB\n",
      "    pexpect-4.6.0              |           py36_0          77 KB\n",
      "    backcall-0.1.0             |           py36_0          19 KB\n",
      "    six-1.11.0                 |           py36_1          21 KB\n",
      "    navigator-updater-0.2.1    |           py36_0         1.2 MB\n",
      "    boto-2.49.0                |           py36_0         1.5 MB\n",
      "    jupyterlab_launcher-0.13.1 |           py36_0          36 KB\n",
      "    markupsafe-1.0             |   py36h1de35cc_1          23 KB\n",
      "    _ipyw_jlab_nb_ext_conf-0.1.0|           py36_0           4 KB\n",
      "    gevent-1.1rc4              |           py36_0         455 KB  bioconda\n",
      "    imageio-2.4.1              |           py36_0         3.3 MB\n",
      "    pytest-3.9.1               |           py36_0         323 KB\n",
      "    service_identity-17.0.0    |   py36h28b3542_0          18 KB\n",
      "    wheel-0.32.2               |           py36_0          35 KB\n",
      "    samtools-1.9               |       h8ee4bcc_1         526 KB  bioconda\n",
      "    glob2-0.4.1                |           py36_0          17 KB  bioconda\n",
      "    networkx-2.2               |           py36_1         2.0 MB\n",
      "    spyder-kernels-0.2.6       |           py36_0          69 KB\n",
      "    incremental-17.5.0         |           py36_0          25 KB\n",
      "    pytest-openfiles-0.3.0     |           py36_0           9 KB\n",
      "    spyder-3.3.1               |           py36_1         2.6 MB\n",
      "    dask-core-0.19.4           |           py36_0         1.2 MB\n",
      "    dask-0.19.4                |           py36_0           3 KB\n",
      "    xlsxwriter-1.1.2           |           py36_0         213 KB\n",
      "    bismark-0.20.0             |                0         1.2 MB  bioconda\n",
      "    bleach-1.4.2               |           py36_0          32 KB  bioconda\n",
      "    pytest-remotedata-0.3.0    |           py36_0          12 KB\n",
      "    jupyter-1.0.0              |           py36_7           6 KB\n",
      "    numexpr-2.6.8              |   py36h1dc9127_0         126 KB\n",
      "    pytest-astropy-0.4.0       |           py36_0           5 KB\n",
      "    sphinx-1.8.1               |           py36_0         1.9 MB\n",
      "    tblib-1.3.2                |   py36hda67792_0          16 KB\n",
      "    beautifulsoup4-4.6.3       |           py36_0         139 KB\n",
      "    typed-ast-1.1.0            |   py36h1de35cc_0         183 KB\n",
      "    mkl_fft-1.0.6              |   py36hb8a8100_0         139 KB\n",
      "    pycosat-0.6.3              |   py36h1de35cc_0         107 KB\n",
      "    simplegeneric-0.8.1        |           py36_2           9 KB\n",
      "    tqdm-4.7.2                 |           py36_0          55 KB  bioconda\n",
      "    future-0.16.0              |           py36_2         668 KB\n",
      "    pillow-5.3.0               |   py36hb68e598_0         550 KB\n",
      "    jupyter_console-6.0.0      |           py36_0          35 KB\n",
      "    pytest-doctestplus-0.1.3   |           py36_0          19 KB\n",
      "    anaconda-project-0.8.2     |           py36_0         478 KB\n",
      "    heapdict-1.0.0             |           py36_2           7 KB\n",
      "    send2trash-1.5.0           |           py36_0          16 KB\n",
      "    scikit-image-0.14.0        |   py36h0a44026_1        23.4 MB\n",
      "    cytoolz-0.9.0.1            |   py36h1de35cc_1         350 KB\n",
      "    sphinxcontrib-1.0          |           py36_1           3 KB\n",
      "    pyasn1-modules-0.0.5       |           py36_0          45 KB  bioconda\n",
      "    idna-2.7                   |           py36_0         132 KB\n",
      "    sqlalchemy-1.2.12          |   py36h1de35cc_0         1.6 MB\n",
      "    backports.shutil_get_terminal_size-1.0.0|           py36_2           8 KB\n",
      "    pathlib2-2.3.2             |           py36_0          32 KB\n",
      "    ipython_genutils-0.2.0     |   py36h241746c_0          39 KB\n",
      "    gmpy2-2.0.8                |   py36h6ef4df4_2         162 KB\n",
      "    numba-0.40.0               |   py36h6440ff4_0         2.5 MB\n",
      "    hyperlink-18.0.0           |           py36_0          62 KB\n",
      "    html5lib-1.0.1             |           py36_0         184 KB\n",
      "    jinja2-2.10                |           py36_0         184 KB\n",
      "    pandas-0.23.4              |   py36h6440ff4_0         9.3 MB\n",
      "    pycodestyle-2.4.0          |           py36_0          59 KB\n",
      "    fastcache-1.0.2            |   py36h1de35cc_2          28 KB\n",
      "    werkzeug-0.14.1            |           py36_0         423 KB\n",
      "    pip-10.0.1                 |           py36_0         1.8 MB\n",
      "    pycurl-7.43.0.2            |   py36hdbc3d79_0          62 KB\n",
      "    perl-5.26.2                |       h4e221da_0        14.8 MB\n",
      "    ruamel_yaml-0.15.46        |   py36h1de35cc_0         228 KB\n",
      "    singledispatch-3.4.0.3     |   py36hf20db9d_0          15 KB\n",
      "    attrs-18.2.0               |   py36h28b3542_0          50 KB\n",
      "    pandocfilters-1.4.2        |           py36_1          13 KB\n",
      "    setuptools-40.4.3          |           py36_0         556 KB\n",
      "    cffi-1.11.5                |   py36h6174b99_1         205 KB\n",
      "    snowballstemmer-1.2.1      |   py36h6c7b616_0          85 KB\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:       239.0 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "    backports.os:                       0.1.1-py36_0                    \n",
      "    bismark:                            0.20.0-0                bioconda\n",
      "    bowtie2:                            2.3.4.3-py36h2d50403_0  bioconda\n",
      "    conda-verify:                       3.1.1-py36_0                    \n",
      "    future:                             0.16.0-py36_2                   \n",
      "    importlib_metadata:                 0.6-py36_0                      \n",
      "    jupyterlab_server:                  0.2.0-py36_0                    \n",
      "    libdeflate:                         1.0-h470a237_0          bioconda\n",
      "    perl:                               5.26.2-h4e221da_0               \n",
      "    samtools:                           1.9-h8ee4bcc_1          bioconda\n",
      "    typed-ast:                          1.1.0-py36h1de35cc_0            \n",
      "\n",
      "The following packages will be REMOVED:\n",
      "\n",
      "    anaconda:                           5.3.0-py37_0                    \n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "    _ipyw_jlab_nb_ext_conf:             0.1.0-py37_0                     --> 0.1.0-py36_0                    \n",
      "    alabaster:                          0.7.11-py37_0                    --> 0.7.12-py36_0                   \n",
      "    anaconda-client:                    1.7.2-py37_0                     --> 1.7.2-py36_0                    \n",
      "    anaconda-navigator:                 1.9.2-py37_0                     --> 1.9.2-py36_0                    \n",
      "    anaconda-project:                   0.8.2-py37_0                     --> 0.8.2-py36_0                    \n",
      "    appdirs:                            1.4.3-py37h28b3542_0             --> 1.4.3-py36h28b3542_0            \n",
      "    appnope:                            0.1.0-py37_0                     --> 0.1.0-py36hf537a9a_0            \n",
      "    appscript:                          1.0.1-py37h1de35cc_1             --> 1.0.1-py36h1de35cc_1            \n",
      "    asn1crypto:                         0.24.0-py37_0                    --> 0.24.0-py36_0                   \n",
      "    astroid:                            2.0.4-py37_0                     --> 2.0.4-py36_0                    \n",
      "    astropy:                            3.0.4-py37h1de35cc_0             --> 3.0.5-py36h1de35cc_0            \n",
      "    atomicwrites:                       1.2.1-py37_0                     --> 1.2.1-py36_0                    \n",
      "    attrs:                              18.2.0-py37h28b3542_0            --> 18.2.0-py36h28b3542_0           \n",
      "    automat:                            0.7.0-py37_0                     --> 0.7.0-py36_0                    \n",
      "    babel:                              2.6.0-py37_0                     --> 2.6.0-py36_0                    \n",
      "    backcall:                           0.1.0-py37_0                     --> 0.1.0-py36_0                    \n",
      "    backports:                          1.0-py37_1                       --> 1.0-py36_1                      \n",
      "    backports.shutil_get_terminal_size: 1.0.0-py37_2                     --> 1.0.0-py36_2                    \n",
      "    beautifulsoup4:                     4.6.3-py37_0                     --> 4.6.3-py36_0                    \n",
      "    bitarray:                           0.8.3-py37h1de35cc_0             --> 0.8.3-py36h1de35cc_0            \n",
      "    bkcharts:                           0.2-py37_0                       --> 0.2-py36h073222e_0              \n",
      "    blaze:                              0.11.3-py37_0                    --> 0.11.3-py36_0                   \n",
      "    bokeh:                              0.13.0-py37_0                    --> 1.0.0-py36_0                    \n",
      "    boto:                               2.49.0-py37_0                    --> 2.49.0-py36_0                   \n",
      "    bottleneck:                         1.2.1-py37h1d22016_1             --> 1.2.1-py36h1d22016_1            \n",
      "    certifi:                            2018.8.24-py37_1                 --> 2018.10.15-py36_0               \n",
      "    cffi:                               1.11.5-py37h6174b99_1            --> 1.11.5-py36h6174b99_1           \n",
      "    chardet:                            3.0.4-py37_1                     --> 3.0.4-py36_1                    \n",
      "    click:                              6.7-py37_0                       --> 7.0-py36_0                      \n",
      "    cloudpickle:                        0.5.5-py37_0                     --> 0.6.1-py36_0                    \n",
      "    clyent:                             1.2.2-py37_1                     --> 1.2.2-py36_1                    \n",
      "    colorama:                           0.3.9-py37_0                     --> 0.4.0-py36_0                    \n",
      "    conda:                              4.5.11-py37_0                    --> 4.5.11-py36_0                   \n",
      "    constantly:                         15.1.0-py37h28b3542_0            --> 15.1.0-py36h28b3542_0           \n",
      "    contextlib2:                        0.5.5-py37_0                     --> 0.5.5-py36hd66e5e7_0            \n",
      "    cryptography:                       2.3.1-py37hdbc3d79_0             --> 2.3.1-py36hdbc3d79_0            \n",
      "    cycler:                             0.10.0-py37_0                    --> 0.10.0-py36hfc81398_0           \n",
      "    cython:                             0.28.5-py37h0a44026_0            --> 0.29-py36h0a44026_0             \n",
      "    cytoolz:                            0.9.0.1-py37h1de35cc_1           --> 0.9.0.1-py36h1de35cc_1          \n",
      "    dask:                               0.19.1-py37_0                    --> 0.19.4-py36_0                   \n",
      "    dask-core:                          0.19.1-py37_0                    --> 0.19.4-py36_0                   \n",
      "    datashape:                          0.5.4-py37_1                     --> 0.5.4-py36_1                    \n",
      "    decorator:                          4.3.0-py37_0                     --> 4.3.0-py36_0                    \n",
      "    defusedxml:                         0.5.0-py37_1                     --> 0.5.0-py36_1                    \n",
      "    distributed:                        1.23.1-py37_0                    --> 1.23.3-py36_0                   \n",
      "    docutils:                           0.14-py37_0                      --> 0.14-py36hbfde631_0             \n",
      "    entrypoints:                        0.2.3-py37_2                     --> 0.2.3-py36_2                    \n",
      "    et_xmlfile:                         1.0.1-py37_0                     --> 1.0.1-py36h1315bdc_0            \n",
      "    fastcache:                          1.0.2-py37h1de35cc_2             --> 1.0.2-py36h1de35cc_2            \n",
      "    filelock:                           3.0.8-py37_0                     --> 3.0.9-py36_0                    \n",
      "    flask:                              1.0.2-py37_1                     --> 1.0.2-py36_1                    \n",
      "    flask-cors:                         3.0.6-py37_0                     --> 3.0.6-py36_0                    \n",
      "    gmpy2:                              2.0.8-py37h6ef4df4_2             --> 2.0.8-py36h6ef4df4_2            \n",
      "    greenlet:                           0.4.15-py37h1de35cc_0            --> 0.4.15-py36h1de35cc_0           \n",
      "    h5py:                               2.8.0-py37h878fce3_3             --> 2.8.0-py36h878fce3_3            \n",
      "    heapdict:                           1.0.0-py37_2                     --> 1.0.0-py36_2                    \n",
      "    html5lib:                           1.0.1-py37_0                     --> 1.0.1-py36_0                    \n",
      "    hyperlink:                          18.0.0-py37_0                    --> 18.0.0-py36_0                   \n",
      "    idna:                               2.7-py37_0                       --> 2.7-py36_0                      \n",
      "    imageio:                            2.4.1-py37_0                     --> 2.4.1-py36_0                    \n",
      "    imagesize:                          1.1.0-py37_0                     --> 1.1.0-py36_0                    \n",
      "    incremental:                        17.5.0-py37_0                    --> 17.5.0-py36_0                   \n",
      "    ipykernel:                          4.9.0-py37_1                     --> 5.1.0-py36h39e3cac_0            \n",
      "    ipython:                            6.5.0-py37_0                     --> 7.0.1-py36h39e3cac_0            \n",
      "    ipython_genutils:                   0.2.0-py37_0                     --> 0.2.0-py36h241746c_0            \n",
      "    ipywidgets:                         7.4.1-py37_0                     --> 7.4.2-py36_0                    \n",
      "    isort:                              4.3.4-py37_0                     --> 4.3.4-py36_0                    \n",
      "    itsdangerous:                       0.24-py37_1                      --> 1.0.0-py36_0                    \n",
      "    jdcal:                              1.4-py37_0                       --> 1.4-py36_0                      \n",
      "    jedi:                               0.12.1-py37_0                    --> 0.13.1-py36_0                   \n",
      "    jinja2:                             2.10-py37_0                      --> 2.10-py36_0                     \n",
      "    jsonschema:                         2.6.0-py37_0                     --> 2.6.0-py36hb385e00_0            \n",
      "    jupyter:                            1.0.0-py37_7                     --> 1.0.0-py36_7                    \n",
      "    jupyter_client:                     5.2.3-py37_0                     --> 5.2.3-py36_0                    \n",
      "    jupyter_console:                    5.2.0-py37_1                     --> 6.0.0-py36_0                    \n",
      "    jupyter_core:                       4.4.0-py37_0                     --> 4.4.0-py36_0                    \n",
      "    jupyterlab:                         0.34.9-py37_0                    --> 0.35.2-py36_0                   \n",
      "    jupyterlab_launcher:                0.13.1-py37_0                    --> 0.13.1-py36_0                   \n",
      "    keyring:                            13.2.1-py37_0                    --> 15.1.0-py36_0                   \n",
      "    kiwisolver:                         1.0.1-py37h0a44026_0             --> 1.0.1-py36h0a44026_0            \n",
      "    lazy-object-proxy:                  1.3.1-py37h1de35cc_2             --> 1.3.1-py36h1de35cc_2            \n",
      "    llvmlite:                           0.24.0-py37hc454e04_0            --> 0.25.0-py36h8c7ce04_0           \n",
      "    locket:                             0.2.0-py37_1                     --> 0.2.0-py36hca03003_1            \n",
      "    lxml:                               4.2.5-py37hef8c89e_0             --> 4.2.5-py36hef8c89e_0            \n",
      "    markupsafe:                         1.0-py37h1de35cc_1               --> 1.0-py36h1de35cc_1              \n",
      "    matplotlib:                         2.2.3-py37h54f8f79_0             --> 3.0.0-py36h54f8f79_0            \n",
      "    mccabe:                             0.6.1-py37_1                     --> 0.6.1-py36_1                    \n",
      "    mistune:                            0.8.3-py37h1de35cc_1             --> 0.8.4-py36h1de35cc_0            \n",
      "    mkl-service:                        1.1.2-py37h6b9c3cc_5             --> 1.1.2-py36h6b9c3cc_5            \n",
      "    mkl_fft:                            1.0.4-py37h5d10147_1             --> 1.0.6-py36hb8a8100_0            \n",
      "    mkl_random:                         1.0.1-py37h5d10147_1             --> 1.0.1-py36h5d10147_1            \n",
      "    more-itertools:                     4.3.0-py37_0                     --> 4.3.0-py36_0                    \n",
      "    mpmath:                             1.0.0-py37_2                     --> 1.0.0-py36_2                    \n",
      "    msgpack-python:                     0.5.6-py37h04f5b5a_1             --> 0.5.6-py36h04f5b5a_1            \n",
      "    multipledispatch:                   0.6.0-py37_0                     --> 0.6.0-py36_0                    \n",
      "    navigator-updater:                  0.2.1-py37_0                     --> 0.2.1-py36_0                    \n",
      "    nbconvert:                          5.4.0-py37_1                     --> 5.4.0-py36_1                    \n",
      "    nbformat:                           4.4.0-py37_0                     --> 4.4.0-py36h827af21_0            \n",
      "    networkx:                           2.1-py37_0                       --> 2.2-py36_1                      \n",
      "    nltk:                               3.3.0-py37_0                     --> 3.3.0-py36_0                    \n",
      "    nose:                               1.3.7-py37_2                     --> 1.3.7-py36_2                    \n",
      "    notebook:                           5.6.0-py37_0                     --> 5.7.0-py36_0                    \n",
      "    numba:                              0.39.0-py37h6440ff4_0            --> 0.40.0-py36h6440ff4_0           \n",
      "    numexpr:                            2.6.8-py37h1dc9127_0             --> 2.6.8-py36h1dc9127_0            \n",
      "    numpy:                              1.15.1-py37h6a91979_0            --> 1.15.3-py36h6a91979_0           \n",
      "    numpy-base:                         1.15.1-py37h8a80b8c_0            --> 1.15.3-py36h8a80b8c_0           \n",
      "    numpydoc:                           0.8.0-py37_0                     --> 0.8.0-py36_0                    \n",
      "    odo:                                0.5.1-py37_0                     --> 0.5.1-py36hc1af34a_0            \n",
      "    olefile:                            0.46-py37_0                      --> 0.46-py36_0                     \n",
      "    packaging:                          17.1-py37_0                      --> 18.0-py36_0                     \n",
      "    pandas:                             0.23.4-py37h6440ff4_0            --> 0.23.4-py36h6440ff4_0           \n",
      "    pandocfilters:                      1.4.2-py37_1                     --> 1.4.2-py36_1                    \n",
      "    parso:                              0.3.1-py37_0                     --> 0.3.1-py36_0                    \n",
      "    partd:                              0.3.8-py37_0                     --> 0.3.9-py36_0                    \n",
      "    path.py:                            11.1.0-py37_0                    --> 11.5.0-py36_0                   \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    pathlib2:                           2.3.2-py37_0                     --> 2.3.2-py36_0                    \r\n",
      "    patsy:                              0.5.0-py37_0                     --> 0.5.0-py36_0                    \r\n",
      "    pep8:                               1.7.1-py37_0                     --> 1.7.1-py36_0                    \r\n",
      "    pexpect:                            4.6.0-py37_0                     --> 4.6.0-py36_0                    \r\n",
      "    pickleshare:                        0.7.4-py37_0                     --> 0.7.5-py36_0                    \r\n",
      "    pillow:                             5.2.0-py37hb68e598_0             --> 5.3.0-py36hb68e598_0            \r\n",
      "    pip:                                10.0.1-py37_0                    --> 10.0.1-py36_0                   \r\n",
      "    pkginfo:                            1.4.2-py37_1                     --> 1.4.2-py36_1                    \r\n",
      "    pluggy:                             0.7.1-py37h28b3542_0             --> 0.8.0-py36_0                    \r\n",
      "    ply:                                3.11-py37_0                      --> 3.11-py36_0                     \r\n",
      "    prometheus_client:                  0.3.1-py37h28b3542_0             --> 0.4.2-py36_0                    \r\n",
      "    prompt_toolkit:                     1.0.15-py37_0                    --> 2.0.6-py36_0                    \r\n",
      "    psutil:                             5.4.7-py37h1de35cc_0             --> 5.4.7-py36h1de35cc_0            \r\n",
      "    ptyprocess:                         0.6.0-py37_0                     --> 0.6.0-py36_0                    \r\n",
      "    py:                                 1.6.0-py37_0                     --> 1.7.0-py36_0                    \r\n",
      "    pyasn1:                             0.4.4-py37h28b3542_0             --> 0.4.4-py36h28b3542_0            \r\n",
      "    pycodestyle:                        2.4.0-py37_0                     --> 2.4.0-py36_0                    \r\n",
      "    pycosat:                            0.6.3-py37h1de35cc_0             --> 0.6.3-py36h1de35cc_0            \r\n",
      "    pycparser:                          2.18-py37_1                      --> 2.19-py36_0                     \r\n",
      "    pycrypto:                           2.6.1-py37h1de35cc_9             --> 2.6.1-py36h1de35cc_9            \r\n",
      "    pycurl:                             7.43.0.2-py37hdbc3d79_0          --> 7.43.0.2-py36hdbc3d79_0         \r\n",
      "    pyflakes:                           2.0.0-py37_0                     --> 2.0.0-py36_0                    \r\n",
      "    pygments:                           2.2.0-py37_0                     --> 2.2.0-py36h240cd3f_0            \r\n",
      "    pylint:                             2.1.1-py37_0                     --> 2.1.1-py36_0                    \r\n",
      "    pyodbc:                             4.0.24-py37h0a44026_0            --> 4.0.24-py36h0a44026_0           \r\n",
      "    pyopenssl:                          18.0.0-py37_0                    --> 18.0.0-py36_0                   \r\n",
      "    pyparsing:                          2.2.0-py37_1                     --> 2.2.2-py36_0                    \r\n",
      "    pyqt:                               5.9.2-py37h655552a_2             --> 5.9.2-py36h655552a_2            \r\n",
      "    pysocks:                            1.6.8-py37_0                     --> 1.6.8-py36_0                    \r\n",
      "    pytables:                           3.4.4-py37h13cba08_0             --> 3.4.4-py36h13cba08_0            \r\n",
      "    pytest:                             3.8.0-py37_0                     --> 3.9.1-py36_0                    \r\n",
      "    pytest-arraydiff:                   0.2-py37h39e3cac_0               --> 0.2-py36h39e3cac_0              \r\n",
      "    pytest-astropy:                     0.4.0-py37_0                     --> 0.4.0-py36_0                    \r\n",
      "    pytest-doctestplus:                 0.1.3-py37_0                     --> 0.1.3-py36_0                    \r\n",
      "    pytest-openfiles:                   0.3.0-py37_0                     --> 0.3.0-py36_0                    \r\n",
      "    pytest-remotedata:                  0.3.0-py37_0                     --> 0.3.0-py36_0                    \r\n",
      "    python-dateutil:                    2.7.3-py37_0                     --> 2.7.3-py36_0                    \r\n",
      "    python.app:                         2-py37_8                         --> 2-py36_9                        \r\n",
      "    pytz:                               2018.5-py37_0                    --> 2018.5-py36_0                   \r\n",
      "    pywavelets:                         1.0.0-py37h1d22016_0             --> 1.0.1-py36h1d22016_0            \r\n",
      "    pyyaml:                             3.13-py37h1de35cc_0              --> 3.13-py36h1de35cc_0             \r\n",
      "    pyzmq:                              17.1.2-py37h1de35cc_0            --> 17.1.2-py36h1de35cc_0           \r\n",
      "    qtawesome:                          0.4.4-py37_0                     --> 0.5.1-py36_1                    \r\n",
      "    qtconsole:                          4.4.1-py37_0                     --> 4.4.2-py36_0                    \r\n",
      "    qtpy:                               1.5.0-py37_0                     --> 1.5.2-py36_0                    \r\n",
      "    rope:                               0.11.0-py37_0                    --> 0.11.0-py36_0                   \r\n",
      "    ruamel_yaml:                        0.15.46-py37h1de35cc_0           --> 0.15.46-py36h1de35cc_0          \r\n",
      "    scikit-image:                       0.14.0-py37h0a44026_1            --> 0.14.0-py36h0a44026_1           \r\n",
      "    scikit-learn:                       0.19.2-py37h4f467ca_0            --> 0.20.0-py36h4f467ca_1           \r\n",
      "    scipy:                              1.1.0-py37h28f7352_1             --> 1.1.0-py36h28f7352_1            \r\n",
      "    seaborn:                            0.9.0-py37_0                     --> 0.9.0-py36_0                    \r\n",
      "    send2trash:                         1.5.0-py37_0                     --> 1.5.0-py36_0                    \r\n",
      "    service_identity:                   17.0.0-py37h28b3542_0            --> 17.0.0-py36h28b3542_0           \r\n",
      "    setuptools:                         40.2.0-py37_0                    --> 40.4.3-py36_0                   \r\n",
      "    simplegeneric:                      0.8.1-py37_2                     --> 0.8.1-py36_2                    \r\n",
      "    singledispatch:                     3.4.0.3-py37_0                   --> 3.4.0.3-py36hf20db9d_0          \r\n",
      "    sip:                                4.19.8-py37h0a44026_0            --> 4.19.8-py36h0a44026_0           \r\n",
      "    six:                                1.11.0-py37_1                    --> 1.11.0-py36_1                   \r\n",
      "    snowballstemmer:                    1.2.1-py37_0                     --> 1.2.1-py36h6c7b616_0            \r\n",
      "    sortedcollections:                  1.0.1-py37_0                     --> 1.0.1-py36_0                    \r\n",
      "    sortedcontainers:                   2.0.5-py37_0                     --> 2.0.5-py36_0                    \r\n",
      "    sphinx:                             1.7.9-py37_0                     --> 1.8.1-py36_0                    \r\n",
      "    sphinxcontrib:                      1.0-py37_1                       --> 1.0-py36_1                      \r\n",
      "    sphinxcontrib-websupport:           1.1.0-py37_1                     --> 1.1.0-py36_1                    \r\n",
      "    spyder:                             3.3.1-py37_1                     --> 3.3.1-py36_1                    \r\n",
      "    spyder-kernels:                     0.2.6-py37_0                     --> 0.2.6-py36_0                    \r\n",
      "    sqlalchemy:                         1.2.11-py37h1de35cc_0            --> 1.2.12-py36h1de35cc_0           \r\n",
      "    statsmodels:                        0.9.0-py37h1d22016_0             --> 0.9.0-py36h1d22016_0            \r\n",
      "    sympy:                              1.2-py37_0                       --> 1.3-py36_0                      \r\n",
      "    tblib:                              1.3.2-py37_0                     --> 1.3.2-py36hda67792_0            \r\n",
      "    terminado:                          0.8.1-py37_1                     --> 0.8.1-py36_1                    \r\n",
      "    testpath:                           0.3.1-py37_0                     --> 0.4.2-py36_0                    \r\n",
      "    toolz:                              0.9.0-py37_0                     --> 0.9.0-py36_0                    \r\n",
      "    tornado:                            5.1-py37h1de35cc_0               --> 5.1.1-py36h1de35cc_0            \r\n",
      "    traitlets:                          4.3.2-py37_0                     --> 4.3.2-py36h65bd3ce_0            \r\n",
      "    unicodecsv:                         0.14.1-py37_0                    --> 0.14.1-py36he531d66_0           \r\n",
      "    wcwidth:                            0.1.7-py37_0                     --> 0.1.7-py36h8c6ec74_0            \r\n",
      "    webencodings:                       0.5.1-py37_1                     --> 0.5.1-py36_1                    \r\n",
      "    werkzeug:                           0.14.1-py37_0                    --> 0.14.1-py36_0                   \r\n",
      "    wheel:                              0.31.1-py37_0                    --> 0.32.2-py36_0                   \r\n",
      "    widgetsnbextension:                 3.4.1-py37_0                     --> 3.4.2-py36_0                    \r\n",
      "    wrapt:                              1.10.11-py37h1de35cc_2           --> 1.10.11-py36h1de35cc_2          \r\n",
      "    xlrd:                               1.1.0-py37_1                     --> 1.1.0-py36_1                    \r\n",
      "    xlsxwriter:                         1.1.0-py37_0                     --> 1.1.2-py36_0                    \r\n",
      "    xlwings:                            0.11.8-py37_0                    --> 0.12.1-py36_0                   \r\n",
      "    zict:                               0.1.3-py37_0                     --> 0.1.3-py36_0                    \r\n",
      "    zope:                               1.0-py37_1                       --> 1.0-py36_1                      \r\n",
      "    zope.interface:                     4.5.0-py37h1de35cc_0             --> 4.5.0-py36h1de35cc_0            \r\n",
      "\r\n",
      "The following packages will be DOWNGRADED:\r\n",
      "\r\n",
      "    bleach:                             2.1.4-py37_0                     --> 1.4.2-py36_0            bioconda\r\n",
      "    conda-build:                        3.15.1-py37_0                    --> 3.10.9-py36_0                   \r\n",
      "    gevent:                             1.3.6-py37h1de35cc_0             --> 1.1rc4-py36_0           bioconda\r\n",
      "    glob2:                              0.6-py37_0                       --> 0.4.1-py36_0            bioconda\r\n",
      "    openpyxl:                           2.5.6-py37_0                     --> 2.4.0-py36_0            bioconda\r\n",
      "    pyasn1-modules:                     0.2.2-py37_0                     --> 0.0.5-py36_0            bioconda\r\n",
      "    python:                             3.7.0-hc167b69_0                 --> 3.6.6-hc167b69_0                \r\n",
      "    requests:                           2.19.1-py37_0                    --> 2.14.2-py36_0                   \r\n",
      "    tqdm:                               4.26.0-py37h28b3542_0            --> 4.7.2-py36_0            bioconda\r\n",
      "    twisted:                            18.7.0-py37h1de35cc_1            --> 17.5.0-py36_0                   \r\n",
      "    urllib3:                            1.23-py37_0                      --> 1.12-py36_0             bioconda\r\n",
      "    xlwt:                               1.3.0-py37_0                     --> 1.2.0-py36h5ad1178_0            \r\n",
      "\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Proceed ([y]/n)? ^C\r\n",
      "\r\n",
      "CondaSystemExit: \r\n",
      "Operation aborted.  Exiting. \r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!conda install -c bioconda bismark"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#make individual reports before making whole summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /Volumes/web/metacarcinus/Pgenerosa/20181011/\n",
    "/Users/strigg/anaconda3/bin/bismark2report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /Volumes/web/metacarcinus/Pgenerosa/20181011/\n",
    "/Users/strigg/anaconda3/bin/bismark2summary \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "##organize files \n",
    "!mkdir /Volumes/web/metacarcinus/Pgenerosa/20181011/OG_bams\n",
    "!mkdir /Volumes/web/metacarcinus/Pgenerosa/20181011/dedup_bams\n",
    "!mkdir /Volumes/web/metacarcinus/Pgenerosa/20181011/dedup_sorted_bams\n",
    "!mkdir /Volumes/web/metacarcinus/Pgenerosa/20181011/BisMeExt\n",
    "!mkdir /Volumes/web/metacarcinus/Pgenerosa/20181011/reports\n",
    "!mv /Volumes/web/metacarcinus/Pgenerosa/20181011/*_pe.bam /Volumes/web/metacarcinus/Pgenerosa/20181011/OG_bams\n",
    "!mv /Volumes/web/metacarcinus/Pgenerosa/20181011/*.deduplicated.bam /Volumes/web/metacarcinus/Pgenerosa/20181011/dedup_bams\n",
    "!mv /Volumes/web/metacarcinus/Pgenerosa/20181011/*.sorted.bam /Volumes/web/metacarcinus/Pgenerosa/20181011/dedup_sorted_bams\n",
    "!mv /Volumes/web/metacarcinus/Pgenerosa/20181011/*.gz /Volumes/web/metacarcinus/Pgenerosa/20181011/BisMeExt\n",
    "!mv /Volumes/web/metacarcinus/Pgenerosa/20181011/*report.txt /Volumes/web/metacarcinus/Pgenerosa/20181011/reports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mv /Volumes/web/metacarcinus/Pgenerosa/20181011/bme.err /Volumes/web/metacarcinus/Pgenerosa/20181011/BisMeExt/\n",
    "!mv /Volumes/web/metacarcinus/Pgenerosa/20181011/dedup.err /Volumes/web/metacarcinus/Pgenerosa/20181011/dedup_bams/\n",
    "!mv /Volumes/web/metacarcinus/Pgenerosa/20181011/*.merged /Volumes/web/metacarcinus/Pgenerosa/20181011/BisMeExt/\n",
    "!mv /Volumes/web/metacarcinus/Pgenerosa/20181011/*.M-bias.txt /Volumes/web/metacarcinus/Pgenerosa/20181011/BisMeExt/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mv /Volumes/web/metacarcinus/Pgenerosa/20181011/*.deduplicated.txt /Volumes/web/metacarcinus/Pgenerosa/20181011/BisMeExt/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
