{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#run multiqc on all EPI samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create data file describing the input data. \n",
    "#Format of the file is 2 column tsv\n",
    "#column 1 = sample name\n",
    "#column 2 = path to bam file\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[31mDay0_amb.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-103_S27.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-103_S27.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-103_S27.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-104_S28.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-104_S28.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-104_S28.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-111_S29.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-111_S29.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-111_S29.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-113_S30.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-113_S30.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-113_S30.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-119_S31.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-119_S31.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-119_S31.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-120_S32.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-120_S32.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-120_S32.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-127_S33.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-127_S33.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-127_S33.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-128_S34.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-128_S34.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-128_S34.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-135_S35.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-135_S35.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-135_S35.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-136_S36.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-136_S36.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-136_S36.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-143_S37.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-143_S37.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-143_S37.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-145_S38.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-145_S38.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-145_S38.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-151_S2.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-151_S2.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-151_S2.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-152_S3.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-152_S3.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-152_S3.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-153_S4.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-153_S4.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-153_S4.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-154_S5.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-154_S5.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-154_S5.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-159_S6.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-159_S6.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-159_S6.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-160_S7.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-160_S7.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-160_S7.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-161_S8.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-161_S8.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-161_S8.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-162_S9.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-162_S9.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-162_S9.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-167_S10.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-167_S10.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-167_S10.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-168_S11.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-168_S11.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-168_S11.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-169_S12.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-169_S12.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-169_S12.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-170_S13.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-170_S13.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-170_S13.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-175_S14.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-175_S14.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-175_S14.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-176_S15.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-176_S15.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-176_S15.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-181_S16.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-181_S16.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-181_S16.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-182_S17.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-182_S17.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-182_S17.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-184_S18.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-184_S18.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-184_S18.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-185_S19.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-185_S19.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-185_S19.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-187_S20.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-187_S20.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-187_S20.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-188_S21.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-188_S21.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-188_S21.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-193_S22.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-193_S22.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-193_S22.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-194_S23.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-194_S23.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-194_S23.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-199_S24.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-199_S24.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-199_S24.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-200_S25.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-200_S25.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-200_S25.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-205_S26.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-205_S26.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-205_S26.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-206_S27.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-206_S27.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-206_S27.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-208_S28.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-208_S28.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-208_S28.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-209_S29.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-209_S29.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-209_S29.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-214_S30.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-214_S30.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-214_S30.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-215_S31.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-215_S31.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-215_S31.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-220_S32.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-220_S32.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-220_S32.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-221_S33.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-221_S33.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-221_S33.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-226_S34.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-226_S34.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-226_S34.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-227_S35.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-227_S35.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-227_S35.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-229_S36.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-229_S36.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-229_S36.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-230_S37.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-230_S37.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-230_S37.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-41_S38.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-41_S38.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-41_S38.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-42_S39.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-42_S39.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-42_S39.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-43_S40.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-43_S40.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-43_S40.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-44_S41.bw\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-44_S41.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mallc_EPI-44_S41.tsv.gz.idx\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax25_cov5x_clst3_rms_results.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax25_cov5x_clst3_rms_results_collapsed.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax25_cov5x_clst3_rms_results_collapsed.tsv.DMR.bed\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax25_cov5x_clst3_rms_results_collapsed.tsv.DMS.bed\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax25_cov5x_rms_results.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv.DMR.bed\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv.DMS.bed\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_collapsed.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_collapsed.tsv.DMR.bed\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_collapsed.tsv.DMS.bed\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold10__49_contigs__length_53961475_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold10__49_contigs__length_53961475_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold10__49_contigs__length_53961475_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold10__49_contigs__length_53961475_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold10__49_contigs__length_53961475_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold10__49_contigs__length_53961475_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold10__49_contigs__length_53961475_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold10__49_contigs__length_53961475_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold11__79_contigs__length_51449921_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold11__79_contigs__length_51449921_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold11__79_contigs__length_51449921_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold11__79_contigs__length_51449921_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold11__79_contigs__length_51449921_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold11__79_contigs__length_51449921_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold11__79_contigs__length_51449921_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold11__79_contigs__length_51449921_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold12__71_contigs__length_50438331_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold12__71_contigs__length_50438331_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold12__71_contigs__length_50438331_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold12__71_contigs__length_50438331_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold12__71_contigs__length_50438331_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold12__71_contigs__length_50438331_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold12__71_contigs__length_50438331_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold12__71_contigs__length_50438331_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold13__52_contigs__length_44396874_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold13__52_contigs__length_44396874_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold13__52_contigs__length_44396874_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold13__52_contigs__length_44396874_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold13__52_contigs__length_44396874_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold13__52_contigs__length_44396874_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold13__52_contigs__length_44396874_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold13__52_contigs__length_44396874_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold14__91_contigs__length_45393038_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold14__91_contigs__length_45393038_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold14__91_contigs__length_45393038_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold14__91_contigs__length_45393038_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold14__91_contigs__length_45393038_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold14__91_contigs__length_45393038_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold14__91_contigs__length_45393038_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold14__91_contigs__length_45393038_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold15__101_contigs__length_47938513_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold15__101_contigs__length_47938513_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold15__101_contigs__length_47938513_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold15__101_contigs__length_47938513_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold15__101_contigs__length_47938513_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold15__101_contigs__length_47938513_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold15__101_contigs__length_47938513_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold15__101_contigs__length_47938513_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold16__33_contigs__length_31980953_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold16__33_contigs__length_31980953_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold16__33_contigs__length_31980953_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold16__33_contigs__length_31980953_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold16__33_contigs__length_31980953_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold16__33_contigs__length_31980953_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold16__33_contigs__length_31980953_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold16__33_contigs__length_31980953_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold17__51_contigs__length_34923512_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold17__51_contigs__length_34923512_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold17__51_contigs__length_34923512_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold17__51_contigs__length_34923512_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold17__51_contigs__length_34923512_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold17__51_contigs__length_34923512_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold17__51_contigs__length_34923512_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold17__51_contigs__length_34923512_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold18__69_contigs__length_27737463_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold18__69_contigs__length_27737463_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold18__69_contigs__length_27737463_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold18__69_contigs__length_27737463_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold18__69_contigs__length_27737463_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold18__69_contigs__length_27737463_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold18__69_contigs__length_27737463_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold18__69_contigs__length_27737463_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold1__77_contigs__length_89643857_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold1__77_contigs__length_89643857_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold1__77_contigs__length_89643857_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold1__77_contigs__length_89643857_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold1__77_contigs__length_89643857_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold1__77_contigs__length_89643857_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold1__77_contigs__length_89643857_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold1__77_contigs__length_89643857_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold2__36_contigs__length_69596280_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold2__36_contigs__length_69596280_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold2__36_contigs__length_69596280_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold2__36_contigs__length_69596280_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold2__36_contigs__length_69596280_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold2__36_contigs__length_69596280_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold2__36_contigs__length_69596280_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold2__36_contigs__length_69596280_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold3__111_contigs__length_57743597_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold3__111_contigs__length_57743597_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold3__111_contigs__length_57743597_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold3__111_contigs__length_57743597_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold3__111_contigs__length_57743597_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold3__111_contigs__length_57743597_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold3__111_contigs__length_57743597_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold3__111_contigs__length_57743597_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold4__129_contigs__length_65288255_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold4__129_contigs__length_65288255_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold4__129_contigs__length_65288255_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold4__129_contigs__length_65288255_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold4__129_contigs__length_65288255_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold4__129_contigs__length_65288255_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold4__129_contigs__length_65288255_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold4__129_contigs__length_65288255_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold5__109_contigs__length_67248332_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold5__109_contigs__length_67248332_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold5__109_contigs__length_67248332_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold5__109_contigs__length_67248332_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold5__109_contigs__length_67248332_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold5__109_contigs__length_67248332_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold5__109_contigs__length_67248332_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold5__109_contigs__length_67248332_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold6__104_contigs__length_61759565_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold6__104_contigs__length_61759565_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold6__104_contigs__length_61759565_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold6__104_contigs__length_61759565_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold6__104_contigs__length_61759565_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold6__104_contigs__length_61759565_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold6__104_contigs__length_61759565_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold6__104_contigs__length_61759565_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold7__69_contigs__length_43120122_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold7__69_contigs__length_43120122_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold7__69_contigs__length_43120122_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold7__69_contigs__length_43120122_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold7__69_contigs__length_43120122_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold7__69_contigs__length_43120122_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold7__69_contigs__length_43120122_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold7__69_contigs__length_43120122_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold8__63_contigs__length_61151155_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold8__63_contigs__length_61151155_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold8__63_contigs__length_61151155_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold8__63_contigs__length_61151155_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold8__63_contigs__length_61151155_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold8__63_contigs__length_61151155_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold8__63_contigs__length_61151155_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold8__63_contigs__length_61151155_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold9__45_contigs__length_38581958_chunk_0.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold9__45_contigs__length_38581958_chunk_1.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold9__45_contigs__length_38581958_chunk_2.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold9__45_contigs__length_38581958_chunk_3.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold9__45_contigs__length_38581958_chunk_4.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold9__45_contigs__length_38581958_chunk_5.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold9__45_contigs__length_38581958_chunk_6.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mamb_AllTimes_DMR250bp_MCmax50_cov5x_clst3_rms_results_for_PGA_scaffold9__45_contigs__length_38581958_chunk_7.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mday10_AllpH_DMR250bp_MCmax25_cov5x_rms_results.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mday10_AllpH_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mday10_AllpH_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv.DMR.bed\u001b[m\u001b[m\r\n",
      "\u001b[31mday10_AllpH_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv.DMS.bed\u001b[m\u001b[m\r\n",
      "\u001b[31mday10_DMRs_minClst2.xml\u001b[m\u001b[m\r\n",
      "\u001b[31mday135_AllpH_DMR250bp_MCmax25_cov5x_rms_results.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mday135_AllpH_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mday135_AllpH_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv.DMR.bed\u001b[m\u001b[m\r\n",
      "\u001b[31mday135_AllpH_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv.DMS.bed\u001b[m\u001b[m\r\n",
      "\u001b[31mday145_AllpH_DMR250bp_MCmax25_cov5x_rms_results.tsv.gz\u001b[m\u001b[m\r\n",
      "\u001b[31mday145_AllpH_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv\u001b[m\u001b[m\r\n",
      "\u001b[31mday145_AllpH_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv.DMR.bed\u001b[m\u001b[m\r\n",
      "\u001b[31mday145_AllpH_DMR250bp_MCmax25_cov5x_rms_results_collapsed.tsv.DMS.bed\u001b[m\u001b[m\r\n",
      "\u001b[34mdedup_bams\u001b[m\u001b[m\r\n",
      "\u001b[34mdestrand_allc_files\u001b[m\u001b[m\r\n",
      "\u001b[34mgenome\u001b[m\u001b[m\r\n",
      "\u001b[34mmerge_allc\u001b[m\u001b[m\r\n",
      "\u001b[34mmerged_allc\u001b[m\u001b[m\r\n",
      "\u001b[31mslurm-1223727.out\u001b[m\u001b[m\r\n",
      "\u001b[31mslurm-1231397.out\u001b[m\u001b[m\r\n",
      "\u001b[31mslurm-1239077.out\u001b[m\u001b[m\r\n",
      "\u001b[31mslurm-1240043.out\u001b[m\u001b[m\r\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams\n"
     ]
    }
   ],
   "source": [
    "cd dedup_bams/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m0807\u001b[m\u001b[m     \u001b[34m0807-003\u001b[m\u001b[m \u001b[34m0807-004\u001b[m\u001b[m \u001b[34m0809-005\u001b[m\u001b[m\r\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      52\r\n"
     ]
    }
   ],
   "source": [
    "!ls -d -1 \"$PWD/\"*/* | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n",
      "\u001b[31m/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\u001b[m\u001b[m\r\n"
     ]
    }
   ],
   "source": [
    "!ls -d -1 \"$PWD/\"*/* "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n"
     ]
    }
   ],
   "source": [
    "!ls */* | sed 's/.*\\///g' | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EPI-175\r\n",
      "EPI-176\r\n",
      "EPI-181\r\n",
      "EPI-182\r\n",
      "EPI-184\r\n",
      "EPI-185\r\n",
      "EPI-187\r\n",
      "EPI-188\r\n",
      "EPI-193\r\n",
      "EPI-194\r\n"
     ]
    }
   ],
   "source": [
    "!ls */* | sed 's/.*\\///g' |sed 's/_S.*//g' | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "!ls */* | sed 's/.*\\///g' |sed 's/_S.*//g' > sample_names.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir multibamQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mv sample_names.txt multibamQC/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EPI-175\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-176\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-181\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-182\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-184\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-185\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-187\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-188\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-193\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n",
      "EPI-194\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\r\n"
     ]
    }
   ],
   "source": [
    "!ls -d -1 \"$PWD/\"*/* | paste multibamQC/sample_names.txt - | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "!ls -d -1 \"$PWD/\"*/* | paste multibamQC/sample_names.txt - > multiBamqc_data_paths.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mv multiBamqc_data_paths.txt multibamQC/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Java memory size is set to 1200M\n",
      "Launching application...\n",
      "\n",
      "QualiMap v.2.2.1\n",
      "Built on 2016-10-03 18:14\n",
      "\n",
      "Selected tool: multi-bamqc\n",
      "Thu Sep 12 12:41:03 PDT 2019\t\tWARNING\tOutput folder already exists, the results will be saved there\n",
      "\n",
      "Running BAM QC on given samples\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 15849240\n",
      "Number of valid reads: 15849240\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 15849240\n",
      "Num mapped first of pair: 7924620\n",
      "Num mapped second of pair: 7924620\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 107\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1434658947\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1431434458\n",
      "numberOfAs: 551983579\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 111\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 20189372\n",
      "Number of valid reads: 20189372\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 20189372\n",
      "Num mapped first of pair: 10094686\n",
      "Num mapped second of pair: 10094686\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 130\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1820216229\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1816075538\n",
      "numberOfAs: 705692335\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 134\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 16420530\n",
      "Number of valid reads: 16420530\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 16420530\n",
      "Num mapped first of pair: 8210265\n",
      "Num mapped second of pair: 8210265\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 108\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1503867602\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1500463597\n",
      "numberOfAs: 578212549\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 112\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 20533044\n",
      "Number of valid reads: 20533044\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 20533044\n",
      "Num mapped first of pair: 10266522\n",
      "Num mapped second of pair: 10266522\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 127\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1866305145\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1862106260\n",
      "numberOfAs: 721855131\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 132\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 12592924\n",
      "Number of valid reads: 12592924\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 12592924\n",
      "Num mapped first of pair: 6296462\n",
      "Num mapped second of pair: 6296462\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 75\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1102113789\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1099636226\n",
      "numberOfAs: 422686640\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 78\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 5532546\n",
      "Number of valid reads: 5532546\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 5532546\n",
      "Num mapped first of pair: 2766273\n",
      "Num mapped second of pair: 2766273\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 37\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 488691009\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 487553464\n",
      "numberOfAs: 186577454\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 38\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 14197810\n",
      "Number of valid reads: 14197810\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 14197810\n",
      "Num mapped first of pair: 7098905\n",
      "Num mapped second of pair: 7098905\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 85\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1267090229\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1264248340\n",
      "numberOfAs: 484580125\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 88\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 12385666\n",
      "Number of valid reads: 12385666\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 12385666\n",
      "Num mapped first of pair: 6192833\n",
      "Num mapped second of pair: 6192833\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 72\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1029922481\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1027675823\n",
      "numberOfAs: 395378404\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 75\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 16341846\n",
      "Number of valid reads: 16341846\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 16341846\n",
      "Num mapped first of pair: 8170923\n",
      "Num mapped second of pair: 8170923\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 97\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1374094675\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1371073218\n",
      "numberOfAs: 528156358\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 101\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 18001892\n",
      "Number of valid reads: 18001892\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 18001892\n",
      "Num mapped first of pair: 9000946\n",
      "Num mapped second of pair: 9000946\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 117\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1652300145\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1648549010\n",
      "numberOfAs: 635208746\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 121\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 8734154\n",
      "Number of valid reads: 8734154\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 8734154\n",
      "Num mapped first of pair: 4367077\n",
      "Num mapped second of pair: 4367077\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 58\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 752099245\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 750376085\n",
      "numberOfAs: 287199805\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 60\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 15001974\n",
      "Number of valid reads: 15001974\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 15001974\n",
      "Num mapped first of pair: 7500987\n",
      "Num mapped second of pair: 7500987\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 93\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1261947264\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1259147873\n",
      "numberOfAs: 482802585\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 96\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 9439830\n",
      "Number of valid reads: 9439830\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 9439830\n",
      "Num mapped first of pair: 4719915\n",
      "Num mapped second of pair: 4719915\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 59\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 768639451\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 766995889\n",
      "numberOfAs: 292937046\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 61\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 13475352\n",
      "Number of valid reads: 13475352\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 13475352\n",
      "Num mapped first of pair: 6737676\n",
      "Num mapped second of pair: 6737676\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 83\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1123891099\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1121468749\n",
      "numberOfAs: 428382900\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 86\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 17282682\n",
      "Number of valid reads: 17282682\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 17282682\n",
      "Num mapped first of pair: 8641341\n",
      "Num mapped second of pair: 8641341\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 102\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1405632699\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1402664230\n",
      "numberOfAs: 536604770\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 106\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 17891018\n",
      "Number of valid reads: 17891018\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 17891018\n",
      "Num mapped first of pair: 8945509\n",
      "Num mapped second of pair: 8945509\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 110\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1559502696\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1556034326\n",
      "numberOfAs: 597891806\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 114\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 17407756\n",
      "Number of valid reads: 17407756\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 17407756\n",
      "Num mapped first of pair: 8703878\n",
      "Num mapped second of pair: 8703878\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 105\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1537917193\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1534514507\n",
      "numberOfAs: 591495789\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 109\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 12785804\n",
      "Number of valid reads: 12785804\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 12785804\n",
      "Num mapped first of pair: 6392902\n",
      "Num mapped second of pair: 6392902\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 74\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1083613036\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1081238463\n",
      "numberOfAs: 414121407\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 77\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 8724714\n",
      "Number of valid reads: 8724714\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 8724714\n",
      "Num mapped first of pair: 4362357\n",
      "Num mapped second of pair: 4362357\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 54\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 760719796\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 758987538\n",
      "numberOfAs: 291363990\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 56\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 9823244\n",
      "Number of valid reads: 9823244\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 9823244\n",
      "Num mapped first of pair: 4911622\n",
      "Num mapped second of pair: 4911622\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 59\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 825754599\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 823926573\n",
      "numberOfAs: 315615238\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 61\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 6876230\n",
      "Number of valid reads: 6876230\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 6876230\n",
      "Num mapped first of pair: 3438115\n",
      "Num mapped second of pair: 3438115\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 44\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 618965528\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 617560650\n",
      "numberOfAs: 236977495\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 46\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 8418704\n",
      "Number of valid reads: 8418704\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 8418704\n",
      "Num mapped first of pair: 4209352\n",
      "Num mapped second of pair: 4209352\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 55\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 728167484\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 726538351\n",
      "numberOfAs: 278921142\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 57\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 12072532\n",
      "Number of valid reads: 12072532\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 12072532\n",
      "Num mapped first of pair: 6036266\n",
      "Num mapped second of pair: 6036266\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 78\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1043744887\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1041433079\n",
      "numberOfAs: 398756007\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 80\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 14564888\n",
      "Number of valid reads: 14564888\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 14564888\n",
      "Num mapped first of pair: 7282444\n",
      "Num mapped second of pair: 7282444\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 95\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1237961007\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1235179151\n",
      "numberOfAs: 475293216\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 98\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 14420554\n",
      "Number of valid reads: 14420554\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 14420554\n",
      "Num mapped first of pair: 7210277\n",
      "Num mapped second of pair: 7210277\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 104\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1307194387\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1304304765\n",
      "numberOfAs: 501321250\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 107\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 16819540\n",
      "Number of valid reads: 16819540\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 16819540\n",
      "Num mapped first of pair: 8409770\n",
      "Num mapped second of pair: 8409770\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 119\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1530560045\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1527104695\n",
      "numberOfAs: 590301082\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 123\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 13399110\n",
      "Number of valid reads: 13399110\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 13399110\n",
      "Num mapped first of pair: 6699555\n",
      "Num mapped second of pair: 6699555\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 106\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1187857248\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1185198318\n",
      "numberOfAs: 455135605\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 109\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 13539046\n",
      "Number of valid reads: 13539046\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 13539046\n",
      "Num mapped first of pair: 6769523\n",
      "Num mapped second of pair: 6769523\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 94\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1222325505\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1219547977\n",
      "numberOfAs: 471192707\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 97\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 9288530\n",
      "Number of valid reads: 9288530\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 9288530\n",
      "Num mapped first of pair: 4644265\n",
      "Num mapped second of pair: 4644265\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 73\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 835074181\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 833189756\n",
      "numberOfAs: 320846819\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 75\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 17984132\n",
      "Number of valid reads: 17984132\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 17984132\n",
      "Num mapped first of pair: 8992066\n",
      "Num mapped second of pair: 8992066\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 121\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1546252988\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1542827822\n",
      "numberOfAs: 594092691\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 125\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 12661652\n",
      "Number of valid reads: 12661652\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 12661652\n",
      "Num mapped first of pair: 6330826\n",
      "Num mapped second of pair: 6330826\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 84\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1099943392\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1097489287\n",
      "numberOfAs: 424205058\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 87\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 12766328\n",
      "Number of valid reads: 12766328\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 12766328\n",
      "Num mapped first of pair: 6383164\n",
      "Num mapped second of pair: 6383164\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 82\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1092267863\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1089851383\n",
      "numberOfAs: 419752058\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 84\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 15270176\n",
      "Number of valid reads: 15270176\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 15270176\n",
      "Num mapped first of pair: 7635088\n",
      "Num mapped second of pair: 7635088\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 98\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1402539648\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1399369062\n",
      "numberOfAs: 541967603\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 101\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 12984772\n",
      "Number of valid reads: 12984772\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 12984772\n",
      "Num mapped first of pair: 6492386\n",
      "Num mapped second of pair: 6492386\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 82\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1179407605\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1176746630\n",
      "numberOfAs: 453623227\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 85\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 12846862\n",
      "Number of valid reads: 12846862\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 12846862\n",
      "Num mapped first of pair: 6423431\n",
      "Num mapped second of pair: 6423431\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 82\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1107275651\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1104839072\n",
      "numberOfAs: 422908161\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 85\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 15594156\n",
      "Number of valid reads: 15594156\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 15594156\n",
      "Num mapped first of pair: 7797078\n",
      "Num mapped second of pair: 7797078\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 107\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1333586212\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1330681192\n",
      "numberOfAs: 512405387\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 110\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 15671672\n",
      "Number of valid reads: 15671672\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 15671672\n",
      "Num mapped first of pair: 7835836\n",
      "Num mapped second of pair: 7835836\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 111\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1465375230\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1462207011\n",
      "numberOfAs: 567981952\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 114\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 23227490\n",
      "Number of valid reads: 23227490\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 23227490\n",
      "Num mapped first of pair: 11613745\n",
      "Num mapped second of pair: 11613745\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 150\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 2129827627\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 2125239505\n",
      "numberOfAs: 825956145\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 156\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 20933644\n",
      "Number of valid reads: 20933644\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 20933644\n",
      "Num mapped first of pair: 10466822\n",
      "Num mapped second of pair: 10466822\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 139\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1909924266\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1905862047\n",
      "numberOfAs: 740834461\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 144\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 19709486\n",
      "Number of valid reads: 19709486\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 19709486\n",
      "Num mapped first of pair: 9854743\n",
      "Num mapped second of pair: 9854743\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 129\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1817616910\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1813714647\n",
      "numberOfAs: 705226236\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 134\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 20827644\n",
      "Number of valid reads: 20827644\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 20827644\n",
      "Num mapped first of pair: 10413822\n",
      "Num mapped second of pair: 10413822\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 140\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1931242379\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1927081899\n",
      "numberOfAs: 750369034\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 145\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 18940212\n",
      "Number of valid reads: 18940212\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 18940212\n",
      "Num mapped first of pair: 9470106\n",
      "Num mapped second of pair: 9470106\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 128\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1752280909\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1748526224\n",
      "numberOfAs: 679724638\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 133\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 16900158\n",
      "Number of valid reads: 16900158\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 16900158\n",
      "Num mapped first of pair: 8450079\n",
      "Num mapped second of pair: 8450079\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 110\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1573917228\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1570532895\n",
      "numberOfAs: 610217397\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 114\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 18600248\n",
      "Number of valid reads: 18600248\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 18600248\n",
      "Num mapped first of pair: 9300124\n",
      "Num mapped second of pair: 9300124\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 128\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1719726175\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1716051202\n",
      "numberOfAs: 666538451\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 133\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 21119008\n",
      "Number of valid reads: 21119008\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 21119008\n",
      "Num mapped first of pair: 10559504\n",
      "Num mapped second of pair: 10559504\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 142\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1885212196\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1881224151\n",
      "numberOfAs: 727470103\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 147\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 20579756\n",
      "Number of valid reads: 20579756\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 20579756\n",
      "Num mapped first of pair: 10289878\n",
      "Num mapped second of pair: 10289878\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 139\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1846563124\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1842692236\n",
      "numberOfAs: 714385066\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 144\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 15792158\n",
      "Number of valid reads: 15792158\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 15792158\n",
      "Num mapped first of pair: 7896079\n",
      "Num mapped second of pair: 7896079\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 104\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1453750443\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1450609854\n",
      "numberOfAs: 563597708\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 108\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 18711866\n",
      "Number of valid reads: 18711866\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 18711866\n",
      "Num mapped first of pair: 9355933\n",
      "Num mapped second of pair: 9355933\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 123\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1729464455\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1725736658\n",
      "numberOfAs: 670578136\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 128\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 12798640\n",
      "Number of valid reads: 12798640\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 12798640\n",
      "Num mapped first of pair: 6399320\n",
      "Num mapped second of pair: 6399320\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 84\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1092141942\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1089679656\n",
      "numberOfAs: 417304426\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 87\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 17152034\n",
      "Number of valid reads: 17152034\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 17152034\n",
      "Num mapped first of pair: 8576017\n",
      "Num mapped second of pair: 8576017\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 105\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1460543909\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1457366123\n",
      "numberOfAs: 556883612\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 109\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 12841222\n",
      "Number of valid reads: 12841222\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 12841222\n",
      "Num mapped first of pair: 6420611\n",
      "Num mapped second of pair: 6420611\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 84\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 1093718890\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 1091312560\n",
      "numberOfAs: 419039444\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 87\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Started processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 417\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 4\n",
      "Processed 50 out of 417 windows...\n",
      "Processed 100 out of 417 windows...\n",
      "Processed 150 out of 417 windows...\n",
      "Processed 200 out of 417 windows...\n",
      "Processed 250 out of 417 windows...\n",
      "Processed 300 out of 417 windows...\n",
      "Processed 350 out of 417 windows...\n",
      "Processed 400 out of 417 windows...\n",
      "Total processed windows:417\n",
      "Number of reads: 8062088\n",
      "Number of valid reads: 8062088\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 8062088\n",
      "Num mapped first of pair: 4031044\n",
      "Num mapped second of pair: 4031044\n",
      "Num singletons: 0\n",
      "Time taken to analyze reads: 58\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 685087833\n",
      "referenceSize: 942353201\n",
      "numberOfSequencedBases: 683589012\n",
      "numberOfAs: 261750813\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 60\n",
      "HTML report created successfully\n",
      "\n",
      "Finished processing /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam\n",
      "\n",
      "BAM QC results are saved to  /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats\n",
      "\n",
      "Running multi-sample BAM QC\n",
      "\n",
      "Checking input paths\n",
      "Loading sample data\n",
      "Creating charts\n",
      "\n",
      "Preparing result report\n",
      "Writing HTML report...\n",
      "HTML report created successfully\n",
      "\n",
      "Finished\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Java HotSpot(TM) 64-Bit Server VM warning: Ignoring option MaxPermSize; support was removed in 8.0\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "/Users/Shared/bioinformatics/qualimap_v2.2.1/qualimap multi-bamqc \\\n",
    "-d multibamQC/multiBamqc_data_paths.txt \\\n",
    "-outdir multibamQC/ \\\n",
    "-r"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#going to redo the multi-bamQC analysis but specifying treatment group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2019-09-12 15:04:58--  https://osf.io/rvy9x/download\n",
      "Resolving osf.io... 35.190.84.173\n",
      "Connecting to osf.io|35.190.84.173|:443... connected.\n",
      "HTTP request sent, awaiting response... 302 FOUND\n",
      "Location: https://files.osf.io/v1/resources/yem8n/providers/osfstorage/5d7167567215100018a00d13?action=download&version=1&direct [following]\n",
      "--2019-09-12 15:04:59--  https://files.osf.io/v1/resources/yem8n/providers/osfstorage/5d7167567215100018a00d13?action=download&version=1&direct\n",
      "Resolving files.osf.io... 35.186.214.196\n",
      "Connecting to files.osf.io|35.186.214.196|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 3212 (3.1K) [text/csv]\n",
      "Saving to: ‘download’\n",
      "\n",
      "download            100%[===================>]   3.14K  --.-KB/s    in 0.002s  \n",
      "\n",
      "2019-09-12 15:05:00 (1.84 MB/s) - ‘download’ saved [3212/3212]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://osf.io/rvy9x/download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m0807\u001b[m\u001b[m       \u001b[34m0807-003\u001b[m\u001b[m   \u001b[34m0807-004\u001b[m\u001b[m   \u001b[34m0809-005\u001b[m\u001b[m   \u001b[31mdownload\u001b[m\u001b[m   \u001b[34mmultibamQC\u001b[m\u001b[m\r\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Date,Sample.ID,Tank,Initial.Treatment,Secondary.Treatment,TimePoint,Homogenization,DNA.Extraction\r\n",
      "16-Mar-16,EPI_41,Time0,Field,NA,Day0,20160912,20161025\r\n",
      "16-Mar-16,EPI_42,Time0,Field,NA,Day0,20160912,20161028\r\n",
      "16-Mar-16,EPI_43,Time0,Field,NA,Day0,20160912,20161025\r\n",
      "16-Mar-16,EPI_44,Time0,Field,NA,Day0,20160912,20161025\r\n",
      "26-Mar-16,EPI_103,Tank11,Low,NA,Day10,20160912,20161028\r\n",
      "26-Mar-16,EPI_104,Tank11,Low,NA,Day10,20160912,20161007\r\n",
      "26-Mar-16,EPI_111,Tank12,Super.Low,NA,Day10,20160912,20161007\r\n",
      "26-Mar-16,EPI_113,Tank12,Super.Low,NA,Day10,20160912,20161007\r\n",
      "26-Mar-16,EPI_119,Tank13,Ambient,NA,Day10,20160909,20161007\r\n"
     ]
    }
   ],
   "source": [
    "!head download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mv download sample.info.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EPI_41 Day0_Field\r\n",
      "EPI_42 Day0_Field\r\n",
      "EPI_43 Day0_Field\r\n",
      "EPI_44 Day0_Field\r\n",
      "EPI_103 Day10_Low\r\n",
      "EPI_104 Day10_Low\r\n",
      "EPI_111 Day10_Super.Low\r\n",
      "EPI_113 Day10_Super.Low\r\n",
      "EPI_119 Day10_Ambient\r\n",
      "EPI_120 Day10_Ambient\r\n"
     ]
    }
   ],
   "source": [
    "!awk -F, 'NR>1{print $2,$6\"_\"$4}' sample.info.csv | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EPI-175\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Low\r\n",
      "EPI-176\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Low\r\n",
      "EPI-181\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Ambient\r\n",
      "EPI-182\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Ambient\r\n",
      "EPI-184\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Ambient\r\n",
      "EPI-185\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Ambient\r\n",
      "EPI-187\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Super.Low\r\n",
      "EPI-188\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Super.Low\r\n",
      "EPI-193\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Low\r\n",
      "EPI-194\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Low\r\n",
      "EPI-199\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Super.Low\r\n",
      "EPI-200\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-003/EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Super.Low\r\n",
      "EPI-205\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Ambient\r\n",
      "EPI-206\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Ambient\r\n",
      "EPI-208\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Low\r\n",
      "EPI-209\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Low\r\n",
      "EPI-214\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Super.Low\r\n",
      "EPI-215\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Super.Low\r\n",
      "EPI-220\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Super.Low\r\n",
      "EPI-221\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Super.Low\r\n",
      "EPI-226\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Ambient\r\n",
      "EPI-227\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Ambient\r\n",
      "EPI-229\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Low\r\n",
      "EPI-230\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807-004/EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day145-Low\r\n",
      "EPI-151\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day135-Ambient\r\n",
      "EPI-152\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day135-Ambient\r\n",
      "EPI-153\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day135-Ambient\r\n",
      "EPI-154\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day135-Ambient\r\n",
      "EPI-159\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day135-Low\r\n",
      "EPI-160\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day135-Low\r\n",
      "EPI-161\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day135-Low\r\n",
      "EPI-162\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day135-Low\r\n",
      "EPI-167\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day135-Super.Low\r\n",
      "EPI-168\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day135-Super.Low\r\n",
      "EPI-169\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day135-Super.Low\r\n",
      "EPI-170\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0807/EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day135-Super.Low\r\n",
      "EPI-103\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day10-Low\r\n",
      "EPI-104\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day10-Low\r\n",
      "EPI-111\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day10-Super.Low\r\n",
      "EPI-113\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day10-Super.Low\r\n",
      "EPI-119\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day10-Ambient\r\n",
      "EPI-120\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day10-Ambient\r\n",
      "EPI-127\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day10-Low\r\n",
      "EPI-128\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day10-Low\r\n",
      "EPI-135\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day10-Ambient\r\n",
      "EPI-136\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day10-Ambient\r\n",
      "EPI-143\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day10-Super.Low\r\n",
      "EPI-145\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day10-Super.Low\r\n",
      "EPI-41\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day0-Field\r\n",
      "EPI-42\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day0-Field\r\n",
      "EPI-43\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day0-Field\r\n",
      "EPI-44\t/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/dedup_bams/0809-005/EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted_stats Day0-Field\r\n"
     ]
    }
   ],
   "source": [
    "!awk -F, 'NR>1{print $2,$6\"_\"$4}' sample.info.csv | \\\n",
    "sed 's/_/-/g' | \\\n",
    "awk 'NR==FNR{a[$1]=$2;next}$1 in a{print $0,a[$1]}' - multibamQC/multiBamqc_data_paths.txt |\\\n",
    "sed 's/sorted\\.bam/sorted_stats/g' \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "!awk -F, 'NR>1{print $2,$6\"_\"$4}' sample.info.csv | \\\n",
    "sed 's/_/-/g' | \\\n",
    "awk 'NR==FNR{a[$1]=$2;next}$1 in a{print $0,a[$1]}' - multibamQC/multiBamqc_data_paths.txt |\\\n",
    "sed 's/sorted\\.bam/sorted_stats/g' \\\n",
    "> multibamQC/multiBamqc_data_paths_sampleCategories.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Java memory size is set to 1200M\n",
      "Launching application...\n",
      "\n",
      "QualiMap v.2.2.1\n",
      "Built on 2016-10-03 18:14\n",
      "\n",
      "Selected tool: multi-bamqc\n",
      "Thu Sep 12 15:19:11 PDT 2019\t\tWARNING\tOutput folder already exists, the results will be saved there\n",
      "\n",
      "Running multi-sample BAM QC\n",
      "\n",
      "Checking input paths\n",
      "Loading sample data\n",
      "Creating charts\n",
      "\n",
      "Preparing result report\n",
      "Writing HTML report...\n",
      "HTML report created successfully\n",
      "\n",
      "Finished\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Java HotSpot(TM) 64-Bit Server VM warning: Ignoring option MaxPermSize; support was removed in 8.0\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "/Users/Shared/bioinformatics/qualimap_v2.2.1/qualimap multi-bamqc \\\n",
    "-d multibamQC/multiBamqc_data_paths_sampleCategories.txt \\\n",
    "-outdir multibamQC/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
