{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# This notebook is for clarifying differences between files used to validate DMRs\n",
    "#### allc files \n",
    "##these are generated by methylpy call-methylation state\n",
    "# command details here: https://github.com/yupenghe/methylpy/blob/methylpy/tutorial/tutorial.md#data-processing\n",
    "#script used to generate allc files here: https://gannet.fish.washington.edu/metacarcinus/mox_jobs/20190822_DMRfindAllEPIsamples.sh\n",
    "# the default settings include filtering for reads with MAPQ >= 30 before DMRs are called\n",
    "\n",
    "#### 5x.bedgraph files\n",
    "##these are files generated by Steven from bismark.cov.gz files where he destranded the data\n",
    "# then filtered for CpGs with >= 5x coverage\n",
    "\n",
    "#### .cov.gz files\n",
    "##these are files generated by bismark_methylation_extractor that are not filtered\n",
    "## by coverage or by MAPQ scores. They are filtered by the score_min threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#With Gannet mounted, navigate to my directory where allc files are"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/web/metacarcinus/Pgenerosa/analyses/20190822\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/web/metacarcinus/Pgenerosa/analyses/20190822/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# for 1 sample (EPI-119) \n",
    "# let's look at the same region (chromosome 2, position 62603614) in all file types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#allc file \n",
    "#columns:\n",
    "#1. scaffold\n",
    "#2. start\n",
    "#3. strand\n",
    "#4. context\n",
    "#5. # Cs methylated\n",
    "#6. # total Cs\n",
    "#7. irrelevant number (only important if phiX control included for conversion rate)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PGA_scaffold2__36_contigs__length_69596280\t62603623\t+\tCAG\t0\t3\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603638\t+\tCGT\t2\t3\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603644\t+\tCGT\t2\t3\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603649\t+\tCCG\t0\t3\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603650\t+\tCGT\t1\t3\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603655\t+\tCTT\t0\t3\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603671\t+\tCGA\t1\t2\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603692\t+\tCAG\t0\t1\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603695\t+\tCGA\t1\t1\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603701\t+\tCTG\t0\t1\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603713\t+\tCGC\t1\t1\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603715\t+\tCCA\t0\t1\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603716\t+\tCAG\t0\t1\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603722\t+\tCAG\t0\t1\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603725\t+\tCGG\t3\t3\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603730\t+\tCGT\t3\t3\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603733\t+\tCCA\t0\t3\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603734\t+\tCAC\t0\t3\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603736\t+\tCAT\t0\t3\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603743\t+\tCAG\t0\t3\t1\n",
      "awk: write error on /dev/stdout\n",
      " input record number 5019123, file \n",
      " source line number 1\n",
      "^C\n"
     ]
    }
   ],
   "source": [
    "!zcat < allc_EPI-119_S31.tsv.gz | \\\n",
    "grep PGA_scaffold2__ | \\\n",
    "awk '{if($2 > 62603614) print $0}' | \\\n",
    "head -20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Steven's 5x bedgraph files\n",
    "#columns:\n",
    "#1. scaffold\n",
    "#2. start\n",
    "#3. stop\n",
    "#4. % methylation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PGA_scaffold2__36_contigs__length_69596280 62603637 62603638 83.3333\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62603643 62603644 85.7143\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62603649 62603650 71.4286\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62603670 62603671 83.3333333333333\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62603724 62603725 100\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62603729 62603730 100\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62603942 62603943 60.0000\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62603972 62603973 60\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62604023 62604024 60\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62604252 62604253 83.3333333333333\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62604279 62604280 83.3333333333333\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62604288 62604289 83.3333333333333\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62604577 62604578 54.5454545454545\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62604617 62604618 75.0000\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62604632 62604633 75.0000\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62604644 62604645 81.2500\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62604654 62604655 77.7778\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62604691 62604692 77.2727\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62604710 62604711 69.2308\r\n",
      "PGA_scaffold2__36_contigs__length_69596280 62604722 62604723 78.5714\r\n",
      "awk: write error on /dev/stdout\r\n",
      " input record number 151880, file \r\n",
      " source line number 1\r\n"
     ]
    }
   ],
   "source": [
    "cat 5xbedgraph/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph | \\\n",
    "grep PGA_scaffold2__ | awk '{if($2 > 62603614) print $0}' | head -20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#original bismark .cov.gz file\n",
    "#columns:\n",
    "#1. scaffold\n",
    "#2. start\n",
    "#3. stop\n",
    "#4. % methylation\n",
    "#5. # Cs methylated\n",
    "#6. # Cs unmethylated "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PGA_scaffold2__36_contigs__length_69596280\t62603638\t62603638\t80\t4\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603639\t62603639\t100\t1\t0\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603644\t62603644\t83.3333333333333\t5\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603645\t62603645\t100\t1\t0\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603650\t62603650\t66.6666666666667\t4\t2\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603651\t62603651\t100\t1\t0\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603671\t62603671\t83.3333333333333\t5\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603686\t62603686\t100\t2\t0\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603695\t62603695\t100\t3\t0\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603713\t62603713\t100\t3\t0\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603725\t62603725\t100\t5\t0\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603730\t62603730\t100\t5\t0\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603922\t62603922\t33.3333333333333\t1\t2\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603923\t62603923\t100\t1\t0\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603943\t62603943\t50\t2\t2\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603944\t62603944\t100\t1\t0\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603967\t62603967\t50\t2\t2\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62603973\t62603973\t60\t3\t2\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62604002\t62604002\t75\t3\t1\n",
      "PGA_scaffold2__36_contigs__length_69596280\t62604006\t62604006\t0\t0\t3\n",
      "awk: write error on /dev/stdout\n",
      " input record number 876570, file \n",
      " source line number 1\n"
     ]
    }
   ],
   "source": [
    "!zcat < /Volumes/web/seashell/bu-mox/scrubbed/0809-005/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz | \\\n",
    "grep PGA_scaffold2__ | awk '{if($2 > 62603614) print $0}' | head -20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# we can see that for the same position, let's look at 62603638\n",
    "# the original .cov.gz file shows it's 80% methylated \n",
    "# Steven's 5x.bedgraph shows it's 83.3% methylated (because it's destranded \n",
    "# and combined 62603638 and 62603639 for a total of 5 methylated Cs and 1 unmethylated C)\n",
    "# allc file shows it's 66.6% methylated with 2 methylated Cs out of 3 total Cs (because\n",
    "# this file filtered out reads with MAPQ < 30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#When I call DMRfind, the differences are based on the allc files filtered for Cs with 5x coverage\n",
    "#so this location which shows 80% methylation in the .cov.gz file and in 5x.bedgraph\n",
    "#would not even be considered in this sample because it has < than 5 reads \n",
    "#that have MAPQ score >=30. \n",
    "\n",
    "#Moral of the story is to validate DMRs, I have to use files that are similarly filtered"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
